src/pages/home/report/ReportActionCompose.js

```diff
@@ -131,20 +132,24 @@ class ReportActionCompose extends React.Component {
 
         this.comment = props.comment;
         this.shouldFocusInputOnScreenFocus = canFocusInputOnScreenFocus();
+        this.selection = {
+            start: props.comment.length,
+            end: props.comment.length,
+        };
 
         this.state = {
             isFocused: this.shouldFocusInputOnScreenFocus,
             isFullComposerAvailable: props.isComposerFullSize,
             textInputShouldClear: false,
             isCommentEmpty: props.comment.length === 0,
             isMenuVisible: false,
-            selection: {
-                start: props.comment.length,
-                end: props.comment.length,
-            },
             maxLines: props.isSmallScreenWidth ? CONST.COMPOSER.MAX_LINES_SMALL_SCREEN : CONST.COMPOSER.MAX_LINES,
-            value: props.comment,
             conciergePlaceholderRandomIndex: _.random(this.props.translate('reportActionCompose.conciergePlaceholderOptions').length - 1),
+
+            // If this is undefined it means we haven't exceeded the max comment length.
+            // If it is a number it means we have exceeded the max comment length and the number is the total length.
+            // We only want to set this value when necessary to avoid re-renders.
```

Comment:
Are these expensive re-renders, or just re-renders of `ExceededCommentLength`? Instead of creating a new state variable, could we just wrap `ExceededCommentLength` with `React.memo` to solve the same problem?

Reply:
yeah, so in general we want to avoid re-rendering the `ReportActionCompose`. That's why I only want to update the comment length state only when necessary.

Another approach would be to make the rendering of `ReportActionCompose` less expensive by further moving parts into sub-components, which we `React.memo`.
However, I in general prefer to save render operations, and running a render on each key stroke is something I'd like to avoid (if it doesn't makes the code too complex)

Reply:
Oh, I see – the `value` used to be stored in state, and we would pass that down to `ExceededCommentLength`. When that changed (i.e: every keystroke), we would re-render `ReportActionCompose` and then `ExceededCommentLength` would update accordingly.

Now, we've made the `TextInput` and uncontrolled component, so need a way to update `ExceededCommentLength` whenever the comment length changes while the comment is too long.

So would another approach (as you've alluded to), be to just pass `this.comment.length` to `ExceededCommentLength` and then wrap it in `React.memo`? Or is that a problem because it wouldn't re-render when we need it to?

Reply:
Overall I guess I'm concerned that we've got a few state values:

- `isCommentEmpty`
- `exceededCommentLength`

That are derived from the same thing – the state of the comment. This creates multiple sources of truth and can lead to confusing and/or hard to maintain code in my experience.

So maybe it's a philosophical difference but I think I might prefer the other approach you've mentioned.

Reply:
Talked with @luacmartins about this a bit more. I think we're happy to take the performance win here, but want to create a follow-up issue to look into refactoring this component, which has become quite complex.

Reply:
100% agreeing on having a follow up and cleaning this further up!